File: 0342-power-of-four/solution.c
bool isPowerOfFour(int n) {
    if(n>0 && ((n&n-1)==0) && ((n-1) == ((((n-1)<<1) & 0xAAAAAAAA) | (((n-1)>>1) & 0x55555555))))
        return true;
    return false;
}

File: 0342-power-of-four/solution.java
class Solution {
    public boolean isPowerOfFour(int n) {
        for(int i=0;i<=15;i++){
            int val = (int) Math.pow(4,i);
            if(val == n) return true;
            if(val > n) return false;
        }
        return false;
    }
}
